src/player.js

```diff
@@ -1808,9 +1836,55 @@ export default class Player extends FakeEventTarget {
   _updateTracks(tracks: Array<Track>): void {
     Player._logger.debug('Tracks changed', tracks);
     this._tracks = tracks;
+    this._tracks = [...tracks, ...this._externalCaptionsHandler.createExternalTracks()];
```

Comment:
why re-assign of this._tracks row after row and not just
` this._tracks = tracks.concat(this._externalCaptionsHandler.createExternalTracks());`
?

Reply:
because the function use player.gertTracks() api, which will return [] if i do it this way. i thought about doind something like that : createExternalTracks(this.tracks) - but i think it looks funny.

Reply:
Not sure I understand you..
U don't need to use gertTracks() since u already get the tracks as an argument. Not u just want to merge those tracks with the externalCaptionsHandler tracks.